python/lsst/meas/extensions/scarlet/deconvolveExposureTask.py

```diff
@@ -0,0 +1,301 @@
+            coadd.image.array[None, :, :],
+            coadd.variance.array[None, :, :],
+            scales=self.config.waveletScales,
+            generation=self.config.waveletGeneration,
+        )
+        # Remove the high frequency wavelets.
+        # This has the effect of preventing high frequency noise
+        # from interfering with the detection of peak positions.
+        wavelets[0] = 0
+        # Reconstruct the image from the remaining wavelet coefficients
+        image = scl.wavelet.starlet_reconstruction(
+            wavelets[:, 0],
+            generation=self.config.waveletGeneration,
+        )
+        # Remove the high frequency wavelets from the PSF.
+        # This is necesary for the image and PSF to have the
+        # same frequency content.
+        psf_wavelets = scl.wavelet.multiband_starlet_transform(
+            psf[None, :, :],
+            scales=self.config.waveletScales,
+            generation=self.config.waveletGeneration,
+        )
+        psf_wavelets[0] = 0
+        psf = scl.wavelet.starlet_reconstruction(
+            psf_wavelets[:, 0],
+            generation=self.config.waveletGeneration,
+        )
+        return image, psf
+
+    def _buildObservation(self, coadd: afwImage.Exposure, band: str):
+        """Build a scarlet lite Observation from an Exposure."""
+        bands = (band,)
+        model_psf = scl.utils.integrated_circular_gaussian(sigma=0.8)
+
+        if self.config.useWavelets:
+            image, psf = self._removeHighFrequencySignal(coadd)
+        else:
+            image = coadd.image.array
+            psf = coadd.getPsf().computeKernelImage(coadd.getBBox().getCenter()).array
+        weights = np.ones_like(coadd.image.array)
+        badPixelMasks = ["SAT", "INTRP", "NO_DATA"]
+        badPixels = coadd.mask.getPlaneBitMask(badPixelMasks)
+        mask = coadd.mask.array & badPixels
+        weights[mask > 0] = 0
+
+        observation = scl.Observation(
+            images=np.array([image.copy()]),
+            variance=np.array([coadd.variance.array.copy()]),
+            weights=np.array([weights]),
+            psfs=np.array([psf]),
+            model_psf=model_psf[None, :, :],
+            convolution_mode="fft",
+            bands=bands,
+            bbox=utils.bboxToScarletBox(coadd.getBBox()),
+        )
+        return observation
+
+    def _initializeModel(
+        self, observation: scl.Observation, peaks: PeakCatalog | None = None
+    ):
+        """Initialize the model for the deconvolution."""
+        if peaks is None:
+            component_peaks = None
+        else:
+            component_peaks = [(peak["i_y"], peak["i_x"]) for peak in peaks]
+
+        # Initialize the model as a single source with a single component:
+        # the entire image.
+        component = scl.models.free_form.FreeFormComponent(
+            bands=observation.bands,
+            model=observation.images.data.copy(),
+            model_bbox=observation.bbox,
```

Comment:
Yes, that's correct. But this is the bounding box of the entire image (none of these algorithms are per blend), since a "component" is really just a deconvolved model of the entire image in this case. So this will be true of any convolution algorithm.

Reply:
I see. The border region won't extend to the inner region of the patch unless the PSF is larger than the overlap then, right?

Reply:
Correct.